cmd/srcd/cmd/sql.go

```diff
@@ -43,98 +45,67 @@ var sqlCmd = &cobra.Command{
 
 		var query string
 		if len(args) == 1 {
-			query = args[0]
+			query = strings.TrimSpace(args[0])
 		}
 
 		client, err := daemon.Client()
 		if err != nil {
 			fatal(err, "could not get daemon client")
 		}
 
-		timeout := 3 * time.Second
-		started := logAfterTimeoutWithServerLogs("this is taking a while, "+
-			"if this is the first time you launch sql client, "+
-			"it might take a few more minutes while we install all the required images",
-			timeout)
-
-		// Might have to pull some images
-		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
-		_, err = client.StartComponent(ctx, &api.StartComponentRequest{
-			Name: components.Gitbase.Name,
-		})
-		started()
-		cancel()
-
-		if err != nil {
-			fatal(err, "could not start gitbase")
-		}
-
-		connReady := logAfterTimeoutWithSpinner("waiting for gitbase to be ready", timeout, 0)
+		startGitbaseWithClient(client)
+		connReady := logAfterTimeoutWithSpinner("waiting for gitbase to be ready", 5*time.Second, 0)
 		err = ensureConnReady(client)
 		connReady()
 		if err != nil {
 			fatal(err, "could not connect to gitbase")
 		}
 
-		if strings.TrimSpace(query) == "" {
-			if err := repl(client); err != nil {
-				log.Fatal(err)
+		// Support piping
+		// TODO(@smacker): not the most optimal solution
+		// it would read all input into memory first and only then send to gitbase
+		// it must be possible to pipe and running mysql-cli with -B flag
+		// but it would change current client behaviour
+		fi, _ := os.Stdin.Stat()
+		if (fi.Mode() & os.ModeCharDevice) == 0 {
+			b, err := ioutil.ReadAll(os.Stdin)
+			if err != nil {
+				fatal(err, "could not read input")
 			}
-			return nil
-		}
 
-		if err := runQuery(client, query); err != nil {
-			log.Fatal(err)
+			query = string(b)
 		}
-		return nil
-	},
-}
-
-func repl(client api.EngineClient) error {
-	rl, err := readline.NewEx(&readline.Config{
-		Prompt: "gitbase> ",
-		Stdin:  os.Stdin,
-		Stderr: os.Stderr,
-		Stdout: os.Stdout,
-	})
-	if err != nil {
-		return err
-	}
-	defer rl.Close()
 
-	for {
-		// read until you get a trailing ';'.
-		var lines []string
-		for {
-			line, err := rl.Readline()
+		resp, exit, err := runMysqlCli(context.Background(), query)
+		if err != nil {
+			fatal(err, "could not run mysql client")
+		}
+		defer resp.Close()
+		defer func() {
+			err := docker.RemoveContainer(components.MysqlCli.Name)
```

Comment:
Why you remove the container? Isn't it better to remove it only on `srcd stop`?

Reply:
that is a very good question. I did it for a reason for sure. But I don't remember this reason anymore... 🤔

Reply:
ahhh. because the container wasn't presented in the `components list` and `stop` wasn't stopping it. After the recent change, I think I don't need to remove it anymore.

Reply:
Ooops. No. We still need it. To show help & `mysql >` with current implementation.
It's possible to run new `mysql` command in the same container each time but it would require more changes. Do you want me to do it in this PR?

Reply:
No I'd consider it as an improvement, thanks for the clarification!

Reply:
Now I remember the original reason. I looked at how web client works. It removes container on exit. So I did the same for cli client.

(moved this comment from a wrong thread)